qa/restart-upgrade/src/test/java/org/opensearch/knn/bwc/AbstractRestartUpgradeTestCase.java

```diff
@@ -0,0 +1,61 @@
+/*
+ *  Copyright OpenSearch Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.bwc;
+
+import org.junit.Before;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.knn.KNNRestTestCase;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+import java.util.Locale;
+
+import static org.opensearch.knn.TestUtils.*;
+
+public abstract class AbstractRestartUpgradeTestCase extends KNNRestTestCase {
+    protected String testIndex;
+
+    @Before
+    protected void setIndex() {
+        // Creating index name by concatenating "knn-bwc-" prefix with test method name
+        // for all the tests in this sub-project
+        testIndex = KNN_BWC_PREFIX + getTestName().toLowerCase(Locale.ROOT);
+    }
+
+    @Override
+    protected final boolean preserveIndicesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected final boolean preserveReposUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected boolean preserveTemplatesUponCompletion() {
+        return true;
+    }
+
+    @Override
+    protected final Settings restClientSettings() {
+        return Settings.builder()
+            .put(super.restClientSettings())
+            // increase the timeout here to 90 seconds to handle long waits for a green
+            // cluster health. the waits for green need to be longer than a minute to
+            // account for delayed shards
+            .put(OpenSearchRestTestCase.CLIENT_SOCKET_TIMEOUT, CLIENT_TIMEOUT_VALUE)
+            .build();
+    }
+
+    protected final boolean isRunningAgainstOldCluster() {
+        return Boolean.parseBoolean(System.getProperty(RESTART_UPGRADE_OLD_CLUSTER));
+    }
+
+    protected final String getBWCVersion() {
```

Comment:
why do we need this method? and how do we handle default null value?

Reply:
We will use it in our future tests. To execute or skip few tests based on the BWCVersion.
```
String bwcVersion = getBWCVersion;
if (bwcVersion == null || bwcVersion.startsWith("1.2") || bwcVersion.startsWith("1.3")) {
     return;
}

Reply:
Can we change this to return Optional as null values are possible?

Reply:
updated